### ISSUE TITLE
Entropy coding

### ISSUE THREAD

**Issue by diff7** [facebookresearch/encodec#42]
## ❓ Questions

Hi, thank you for a great work.

(I) 
I could not figure out necessity of predicting codebook logits  via transformer. 
Why could not we use empirical distribution of codebook usage ( frequencies ) over a validation set?  
I feel like I am missing something here.

(II)
Also, [3] and  your work show that Entropy coding does not improve much.  While [2] demonstrates significantly better results (excluding latency most likely). Also, most of the literature on Neural Image codecs shows that most of the gains are achieved from Entropy coding. Any comments on that? Do you think  different implementation is required  to obtain higher compression or it is not that important for Audio domain? 

(III)
Also, I am writing on behalf of a small non commercial, independent research group. We are interested to work in this direction further, if there are some possibilities to collaborate that would be wonderful.  We have some ideas and resources to test them.  Your expertise could help us to save some time.

@adefossez. 
Actually, I am sort of a fan of yours since speech de-noising paper [1]. RNNs within auto-encoders looks like a signature move, not sure if it is your idea though. ))

[1]  Real Time Speech Enhancement in the Waveform Domain
[2]  LMCODEC: A LOW BITRATE SPEECH CODEC WITH CAUSAL TRANSFORMER MODELS
[3] SoundStream: An End-to-End Neural Audio Codec

**Comment by adefossez** [comment]
(1) using the marginal distribution will lead to worse performance than conditional distribution as the entropy of x_t is necessarily more than x_t | x_{t-1}, x_{t-2} ...

(2) [2] does not transmit some of the codes but instead "synthetise" them through a LM. we only use the LM for entropic coding. the key difference is that the generation step might hallucinate some part of the audio, while in our case the EC approach is garanteed to give the same output as if all the codes were transmitted (in fact they are all transmited).

(3) thanks for the proposal and kind words, at the moment I am not looking for more collaborations though ! best luck for your projects.

**Comment by diff7** [comment]
Thanks a lot for your response, very much appreciated!
